#### Estimation and Confidence Interval

Scenario

A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.


a. Build 99% Confidence Interval Using Sample Standard Deviation

In [37]:
## 1. Here we will use t- test because the sample size is leass than 30

In [17]:
import numpy as np
import statistics
from scipy import stats

In [1]:
sample_data = 1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

In [25]:
#calculate mean
mean = round(statistics.mean(sample_data),2)
print("Sample Mean:",mean)

#calculate standard deviation
std = round(statistics.stdev(sample_data),2)
print("Sample Standard Deviation:",std)

#declare sample size
sample_size = 15
print("Sample Size:", sample_size)

#define confidence interval
confidence_interval = 0.99
print("Confidence Interval:", confidence_interval)

## Degree of Freedom
df = sample_size - 1
print("Degree of Freedom:",df)

Sample Mean: 1.24
Sample Standard Deviation: 0.19
Sample Size: 15
Confidence Interval: 0.99
Degree of Freedom: 14


In [21]:
## calculate significance level
alpha = (1 - confidence_interval)/2
alpha

0.0050000000000000044

In [32]:
## Calculate t-value
t_value = stats.t.ppf(1-alpha,df)
print(t_value)

2.976842734370834


In [33]:
#calculate margin of error using formula
margin_of_error = t_value*(std/np.sqrt(sample_size))
print(f"Margin of error:{margin_of_error}")

Margin of error:0.146037322903626


In [42]:
lower_limit = (mean - margin_of_error).round(2)
upper_limit = (mean + margin_of_error).round(2)

print ((f"Confidence Interval:{lower_limit},{upper_limit}"))

Confidence Interval:1.09,1.39


Conclusion (based on t_value):
- We are 99% confident that the mean number of characters printed before the print head fails is between 1.09 and 1.39 millions of characters.

b. Build 99% Confidence Interval Using Known Population Standard Deviation

In [ ]:
## Here population standard deviation is given, hence we will use z-test

In [49]:
population_stdev = 0.2

In [ ]:
mean = round(statistics.mean(sample_data),2)
print("Sample Mean:",mean)

#define confidence interval
confidence_interval = 0.99
print("Confidence Interval:", confidence_interval)

In [43]:
## calculate significance level
alpha = (1 - confidence_interval)/2
alpha

0.0050000000000000044

In [47]:
# calculate z-value
z_value = stats.norm.ppf((1-alpha))
z_value

np.float64(2.5758293035489004)

In [50]:
#calculate margin of error using formula
margin_of_error = z_value*(population_stdev/np.sqrt(sample_size))
print(f"Margin of error:{margin_of_error}")

Margin of error:0.13301525327090588


In [51]:
lower_limit = (mean - margin_of_error).round(2)
upper_limit = (mean + margin_of_error).round(2)

print ((f"Confidence Interval:{lower_limit},{upper_limit}"))

Confidence Interval:1.11,1.37


Conclusion (based on z_value):
- We are 99% confident that the mean number of characters printed before the print head fails is between 1.11 and 1.37 millions of characters.